internal/api/websockets/marketdata/marketdata_server.go

```diff
@@ -0,0 +1,83 @@
+package websockets
+
+import (
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/websocket"
+)
+
+type WebSocketServer struct {
+	clients    map[*websocket.Conn]bool
+	broadcast  chan []byte
+	register   chan *websocket.Conn
+	unregister chan *websocket.Conn
+	mutex      sync.Mutex
+}
+
```

Comment:
```suggestion
type Client struct {
	conn *websocket.Conn
	send chan []byte
}
```

You need a Client struct to manage individual client connections and their channels.